src/libstd/io/buffered.rs

```diff
@@ -257,6 +257,41 @@ impl<R: Seek> BufReader<R> {
         }
         self.seek(SeekFrom::Current(offset)).map(drop)
     }
+
+    /// Returns the current seek position from the start of the stream.
+    ///
+    /// The value returned is equivalent to `self.seek(SeekFrom::Current(0))`
+    /// but does not flush the internal buffer. Due to this optimization the
+    /// function does not guarantee that calling `.into_inner()` immediately
+    /// afterwards will yield the underlying reader at the same position. Use
+    /// [`BufReader::seek`] instead if you require that guarantee.
+    ///
+    /// [`BufReader::seek`]: struct.BufReader.html#method.seek
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// #![feature(seek_convenience)]
+    /// use std::{
+    ///     io::{self, BufRead, BufReader},
+    ///     fs::File,
+    /// };
+    ///
+    /// fn main() -> io::Result<()> {
+    ///     let mut f = BufReader::new(File::open("foo.txt")?);
+    ///
+    ///     let before = f.stream_position()?;
+    ///     f.read_line(&mut String::new())?;
+    ///     let after = f.stream_position()?;
+    ///
+    ///     println!("The first line was {} bytes long", after - before);
+    ///     Ok(())
+    /// }
+    /// ```
+    #[unstable(feature = "seek_convenience", issue = "59359")]
+    pub fn stream_position(&mut self) -> io::Result<u64> {
+        self.inner.stream_position().map(|pos| pos - (self.cap - self.pos) as u64)
+    }
```

Comment:
I think you meant to put that into the `impl<R: Seek> Seek for BufReader<R>` block instead?

Reply:
Oh, yes indeed. Shall I add a new commit to fix this or squash the change into the existing one?

Reply:
I'd prefer squashing/amending in this case, but however you like!